prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -4951,16 +4973,23 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             location,
                             vir::AssignKind::Copy,
                         )?;
-                        // Initialize the constant
-                        let const_val = self.encoder
-                            .encode_const_expr(ty, &val)
-                            .with_span(span)?;
-                        // Initialize value of lhs
-                        stmts.push(vir::Stmt::Assign(
-                            lhs.clone().field(field),
-                            const_val,
-                            vir::AssignKind::Copy,
-                        ));
+                        let is_str = match ty.kind() {
+                            ty::TyKind::Ref(_, inner, _) => inner.is_str(),
```

Comment:
This should check that the string is `'static`.

Reply:
I think the same thing applies here. If we already know that it's a constant, shouldn't it certainly be `'static`?